.github/workflows/ci.yml

```diff
@@ -69,20 +70,29 @@ jobs:
       # XXX this should be instant because all the stages were already built in the steps above
       # but it's re-building the last stage. This is deemed good enough for now.
       # see https://github.com/moby/moby/issues/34715
-      - name: Build and push complete image
+      - name: Build and push native-only image
         uses: docker/build-push-action@v5
+        if: "github.event_name == 'pull_request' || github.event_name == 'merge_group'"
         with:
           context: .
           # push to registry on every repo push. A PR #2 will push with tag `pr-2` and `main` will have tag `main`.
           # See https://github.com/docker/metadata-action?tab=readme-ov-file#basic.
           push: true
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
-      - name: notify on failure
-        if: failure() && github.event_name != 'pull_request'
-        uses: ./.github/actions/notify-status
+
+      - name: Set up QEMU for cross-platform builds
+        uses: docker/setup-qemu-action@v3
+        if: "!(github.event_name == 'pull_request' || github.event_name == 'merge_group')"
+
+      - name: Build and push all platform images
+        uses: docker/build-push-action@v5
+        if: "!(github.event_name == 'pull_request' || github.event_name == 'merge_group')"
         with:
-          webhook: ${{ secrets.SLACK_WEBHOOK_URL }}
-          from: ${{ secrets.NOTIFY_EMAIL_FROM }}
-          to: ${{ secrets.NOTIFY_EMAIL_TO }}
-          password: ${{ secrets.NOTIFY_EMAIL_PASSWORD }}
+          context: .
+          platforms: linux/amd64,linux/arm64/v8
```

Comment:
Why not have a single unconditional `docker/build-push-action` step, and make the `platforms` value based on the event ?